pkg/charts/linkerd2/values.go

```diff
@@ -31,6 +31,7 @@ type (
 		ControllerUID                int64               `json:"controllerUID"`
 		EnableH2Upgrade              bool                `json:"enableH2Upgrade"`
 		EnablePodAntiAffinity        bool                `json:"enablePodAntiAffinity"`
+	    NodeAffinity                 interface{}         `json:"nodeAffinity"`
```

Comment:
I think you can use `map[string]string` instead. In order to check, try using this new setting through the CLI, like `linkerd install --set nodeAffinity=...`

Reply:
Yep, you are right.

Running 
```bash
./bin/linkerd install \
--set "nodeAffinity.requiredDuringSchedulingIgnoredDuringExecution.nodeSelectorTerms[0].matchExpressions[0].key=kubernetes.io/os" \
--set "nodeAffinity.requiredDuringSchedulingIgnoredDuringExecution.nodeSelectorTerms[0].matchExpressions[0].operator=In" \
--set "nodeAffinity.requiredDuringSchedulingIgnoredDuringExecution.nodeSelectorTerms[0].matchExpressions[0].values[0]=linux" 
```

produces

```yaml
      affinity:
        nodeAffinity:
          requiredDuringSchedulingIgnoredDuringExecution:
            nodeSelectorTerms:
            - matchExpressions:
              - key: kubernetes.io/os
                operator: In
                values:
                - linuxk
```